scripts/dockerimages/scripts/rancheros-install

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+set -e
+
+usage()
+{
+cat <<EOF
+Usage: 
+  rancheros-install [options]
+Options:
+  -c cloud-config file
+      needed for SSH keys.
+  -d device
+  -f [ DANGEROUS! Data loss can happen ] partition/format without prompting
+  -p partition disk.
+        This will run fdisk and create one large ext4 partition.
+  -t install-type:
+        virtualbox-iso
+        amazon-ebs
+        rancher-upgrade
+  -v os-installer version.
+  -h print this
+EOF
+}
+
+while getopts "c:d:fpt:v:h" OPTION
+do
+    case $OPTION in
+        c) CLOUD_CONFIG="$OPTARG" ;;
+        d) DEVICE="$OPTARG" ;;
+        f) FORCE_INSTALL="true" ;;
+        p) PARTITION_FLAG="true" ;;
+        t) INSTALL_TYPE="${OPTARG}" ;;
+        v) INSTALLER_VERSION="$OPTARG" ;;
+        h) usage; exit ;;
+        *) exit 1 ;;
+    esac
+done
+
+if [ "$(whoami)" != "root" ]; then
+    echo "Please run as root." 1>&2
+    exit 1
+fi
+
+if [[ -z "${INSTALL_TYPE}" ]]; then
+    echo "$0: No install type specified, -t required to install" 1>&2
+    exit 1
+fi
+
+if [ -z "${CLOUD_CONFIG}" ] && [ "${INSTALL_TYPE}" != "amazon-ebs" ]; then
+    echo "$0: called without cloud config. Can not proceed without -c" 1>&2
+    exit 1
+fi
+
+if [[ ! -z "${CLOUD_CONFIG}" ]]; then
+    cp ${CLOUD_CONFIG} /opt/user_config.yml
+    CLOUD_INIT_DATA="/opt/user_config.yml"
+fi
+
+if [ "${FORCE_INSTALL}" != "true" ] && [ "${INSTALL_TYPE}" != "rancher-upgrade" ]; then
+    echo "This will wipe all data on ${DEVICE}!"
```

Comment:
Do you want to print what you are about to do before you prompt?  Like "Partition: true\nDevice:/dev/sda"